src/btree/bt_compact.c

```diff
@@ -187,6 +215,9 @@ __wt_compact(WT_SESSION_IMPL *session)
 
 		session->compact_state = WT_COMPACT_SUCCESS;
 		WT_STAT_DATA_INCR(session, btree_compact_rewrite);
+
+		if (WT_VERBOSE_ISSET(session, WT_VERB_COMPACT_PROGRESS))
+			__compact_progress(session);
```

Comment:
I think this should move to the code block that contains `__wt_session_compact_check_timeout` - it's similar code and calling this progress function less often is appealing.